presto-common/src/main/java/com/facebook/presto/common/block/Block.java

```diff
@@ -205,6 +205,15 @@ default long getLogicalSizeInBytes()
      */
     long getRegionSizeInBytes(int position, int length);
 
+    /**
+     * Returns the size of {@code block.getRegion(position, length)}.
+     * The method can be expensive. Do not use it outside an implementation of Block.
+     */
+    default long getRegionLogicalSizeInBytes(int position, int length)
```

Comment:
@mbasmanova For leaf blocks (ie. non Array/Map/Row/Dict/RLE blocks), the logicalSizeInBytes is the same as sizeInBytes. See the following code:
```
/**
     * Returns the size of the block contents, regardless of internal representation.
     * The same logical data values should always have the same size, no matter
     * what block type is used or how they are represented within a specific block.
     *
     * This can differ substantially from {@link #getSizeInBytes} for certain block
     * types. For RLE, it will be {@code N} times larger. For dictionary, it will be
     * larger based on how many times dictionary entries are reused.
     */
    default long getLogicalSizeInBytes()
    {
        return getSizeInBytes();
    }
```
Similarly, regional logical size for leaf blocks is the same as the regional size. We have default implementation here so that we don't have to implement the same thing for all leaf blocks.